internal/olm/operator/bundle/install.go

```diff
@@ -98,3 +168,169 @@ func (i *Install) setup(ctx context.Context) error {
 
 	return nil
 }
+
+func (f *FBCContext) generateFBCContent(ctx context.Context, bundleImage, indexImage string) (string, error) {
+	log.Infof("Creating a File-Based Catalog of the bundle %q", bundleImage)
+	// generate a File-Based Catalog representation of the bundle image
+	bundleDeclcfg, err := f.createFBC(ctx)
+	if err != nil {
+		return "", fmt.Errorf("error creating a File-Based Catalog with image %q: %v", bundleImage, err)
+	}
+
+	declcfg := &declarativeconfig.DeclarativeConfig{
+		Bundles:  []declarativeconfig.Bundle{bundleDeclcfg.Bundle},
+		Packages: []declarativeconfig.Package{bundleDeclcfg.Package},
+		Channels: []declarativeconfig.Channel{bundleDeclcfg.Channel},
+	}
+
+	if indexImage != registry.DefaultIndexImage { // non-default index image was specified.
+		// since an index image is specified, the bundle image will be added to the index image.
+		// addBundleToIndexImage will ensure that the bundle is not already present in the index image and error out if it does.
+		declcfg, err = addBundleToIndexImage(ctx, indexImage, declcfg)
+		if err != nil {
+			return "", fmt.Errorf("error adding bundle image %q to index image %q: %v", bundleImage, indexImage, err)
+		}
+	}
+
+	// validate the generated File-Based Catalog
+	if err = validateFBC(declcfg); err != nil {
+		return "", fmt.Errorf("error validating the generated FBC: %v", err)
+	}
+
+	// convert declarative config to string
+	content, err := stringifyDecConfig(declcfg)
+	if err != nil {
+		return "", fmt.Errorf("error converting the declarative config to string: %v", err)
+	}
+
+	if content == "" {
+		return "", errors.New("file based catalog contents cannot be empty")
+	}
+
+	log.Infof("Generated a valid File-Based Catalog")
+
+	return content, nil
+}
+
+// addBundleToIndexImage adds the bundle to an existing index image if the bundle is not already present in the index image.
+func addBundleToIndexImage(ctx context.Context, indexImage string, bundleDeclConfig *declarativeconfig.DeclarativeConfig) (*declarativeconfig.DeclarativeConfig, error) {
```

Comment:
Should this function accept `bundleDeclcfg` instead of `declarativeconfig.DeclarativeConfig`? Seems like we're assuming we have exactly one each of package/channel/bundle.

Reply:
It's probably better to leave it as is, because we are calling `validateFBC` function  right after this call and that function expects the type `declarativeconfig.DeclarativeConfig` to convert to a model. So this conversion from `bundleDeclcfg` to `declarativeconfig.DeclarativeConfig` needs to happen anyways. WDYT?

Reply:
Ignore my previous comment. :)

I've [modified it](https://github.com/operator-framework/operator-sdk/pull/5671/commits/21dc19831a1a23a0df944252514f5d5449accb08) based on your suggestion and now converting to the `DeclarativeConfig` type just before the validation checks. I think this makes sense as the intent is to have the function expect a bundle declarative config with exactly one of bundles/packages/channels.